install/tools/setupMeshEx.sh

```diff
@@ -131,23 +131,37 @@ function istio_provision_certs() {
 #
 # Expected to be run from the release directory (ie istio-0.2.8/ or istio/)
 function istioBootstrapVM() {
- local DESTINATION=${1}
-
- local SA=${2:-${SERVICE_ACCOUNT:-default}}
- local NS=${3:-${SERVICE_NAMESPACE:-}}
- echo "Making certs for service account $SA (namespace $NS)"
- istio_provision_certs $SA $NS
-
- # Copy deb, helper and config files
- istioCopy $DESTINATION \
-   kubedns \
-   *.pem \
-   cluster.env \
-   istio.VERSION \
-   install/tools/setupIstioVM.sh
-
- # Run the setup script.
- istioRun $DESTINATION "sudo bash -c -x ./setupIstioVM.sh"
+  local DESTINATION=${1}
+  local SA=${2:-${SERVICE_ACCOUNT:-default}}
+  local NS=${3:-${SERVICE_NAMESPACE:-}}
+
+  DEFAULT_SCRIPT="install/tools/setupIstioVM.sh"
+  SETUP_ISTIO_VM_SCRIPT=${SETUP_ISTIO_VM_SCRIPT:-${DEFAULT_SCRIPT}}
+  echo "Making certs for service account $SA (namespace $NS)"
+  istio_provision_certs $SA $NS
+
+  for i in {1..10}; do
+    # Copy deb, helper and config files
+    istioCopy $DESTINATION \
+      kubedns \
+      *.pem \
+      cluster.env \
+      istio.VERSION \
+      ${SETUP_ISTIO_VM_SCRIPT}
+
+    if [[ $? -ne 0 ]]; then
+      echo "scp failed, retry in 10 sec"
```

Comment:
when/why does this fail ?
should we be able to do a liveness (ssh date) or such instead of retrying everything ?

Reply:
I totally agree with you but GCP has been flaky where the scp could fail (shown from log) even after we have successfully ssh into the vm. Then the retry is required since we need all the files copied to be able to test the vm.